src/kernels/jupyter/connection/serverUriStorage.ts

```diff
@@ -372,16 +370,7 @@ class NewStorage {
                     displayName: item.displayName || uri,
                     provider: item.serverHandle
                 };
-                if (!validate) {
-                    entries.push(server);
-                    return;
-                }
-                try {
-                    await this.jupyterPickerRegistration.getJupyterServerUri(item.serverHandle, true);
```

Comment:
This is only required in one calling place, removed this and move the validation to the calling code.
Thereby removing a cycling reference and more layers.
This also makes listing of Jupyter Server much faster, as we no longer validate each server, which in turn waits for all related extensions to completely activate.